lib/process-services-cloud/src/lib/task/task-filters/components/task-assignment-filter/task-assignment-filter.component.spec.ts

```diff
@@ -54,34 +70,135 @@ describe('EditTaskFilterCloudComponent', () => {
     });
 
-    it('should show the candidate groups', () => {
-        component.assignmentType = AssignmentType.CANDIDATE_GROUPS;
-        fixture.detectChanges();
-        const candidateGroups = fixture.debugElement.nativeElement.querySelector('.adf-group-cloud-filter');
+    it('should emit the current user info when assignment is the current user', () => {
+        spyOn(identityUserService, 'getCurrentUserInfo').and.returnValue(mockIdentityUsers[0]);
+        spyOn(component.assignedUsersChange, 'emit');
+
+        selectAssignmentType(AssignmentType.CURRENT_USER);
+
+        expect(component.assignedUsersChange.emit).toHaveBeenCalledWith([mockIdentityUsers[0]]);
+    });
+
+    it('should show the CANDIDATE_GROUPS input', () => {
+        selectAssignmentType(AssignmentType.CANDIDATE_GROUPS);
+
+        const candidateGroups = fixture.debugElement.query(By.css('[data-automation-id="adf-group-cloud-candidate-groups-filter"]'));
+        expect(component.candidateGroups.length).toEqual(0);
         expect(candidateGroups).toBeTruthy();
     });
 
+    it('should show the ASSIGNED_TO input', () => {
+        selectAssignmentType(AssignmentType.ASSIGNED_TO);
+
+        const candidateGroups = fixture.debugElement.query(By.css('[data-automation-id="adf-group-cloud-assigned-to-filter"]'));
+        expect(component.assignedUsers.length).toEqual(0);
+        expect(candidateGroups).toBeTruthy();
+    });
+
+    describe('status input change', () => {
+        it('should CREATED status set assignment type to UNASSIGNED', () => {
+            const createdStatusChange = new SimpleChange(null, 'CREATED', true);
+            component.ngOnChanges({status: createdStatusChange});
+
+            expect(component.assignmentType).toEqual(AssignmentType.UNASSIGNED);
+        });
+
+        it('should ASSIGNED status set assignment type to ASSIGNED_TO', () => {
+            const createdStatusChange = new SimpleChange(null, 'ASSIGNED', true);
+            component.ngOnChanges({status: createdStatusChange});
+
+            expect(component.assignmentType).toEqual(AssignmentType.ASSIGNED_TO);
+        });
+
+        it('should empty status set assignment type to NONE', () => {
+            const createdStatusChange = new SimpleChange(null, '', true);
+            component.ngOnChanges({status: createdStatusChange});
+
+            expect(component.assignmentType).toEqual(AssignmentType.NONE);
+        });
+    });
+
+    describe('set initial assignment type', () => {
+        it('should set assignment type to ASSIGNED_TO if initial assignedUsers exists', () => {
+            component.taskFilterProperty = {
+                key: 'assignment',
+                label: 'mock-filter',
+                value: { assignedUsers: mockIdentityUsers },
+                type: 'assignment',
+                attributes: { assignedUsers: 'assignedUsers', candidateGroups: 'candidateGroups'}
+            };
+            component.ngOnInit();
```

Comment:
is there a way to avoid force calling ngOnInit and do fixture.detectChanges() instead?

Reply:
changed